runtime/gc_base/RootScanner.cpp

```diff
@@ -887,6 +887,17 @@ MM_RootScanner::scanRoots(MM_EnvironmentBase *env)
 #endif /* J9VM_GC_FINALIZATION */
 	scanJNIGlobalReferences(env);
 
+	if (_jniWeakGlobalReferencesTableAsRoot) {
+		/*
+		 * Usually JNI Weak Global References table is scanned at Clearable phase.
+		 * However in the case of Concurrent Scavenger it is required to be scanned as a hard root
+		 * The reason for this VM uses elements of table without calling a Read Barrier,
+		 * so JNI Weak Global References table should be treated as a hard root until VM code is fixed
+		 * and Read Barrier is called for each single object.
+		 */
+		scanJNIWeakGlobalReferences(env);
+	}
+
```

Comment:
most of this comment I'd actually put Scavenger Root scanner constructor.

Reply:
ok